winpr/libwinpr/synch/barrier.c

```diff
@@ -165,6 +167,8 @@ BOOL WINAPI DeleteSynchronizationBarrier(LPSYNCHRONIZATION_BARRIER lpBarrier)
 	if (!pBarrier)
 		return TRUE;
 
+	while (InterlockedCompareExchange(&pBarrier->count, 0, 0) != 0);
+
```

Comment:
For a high count this might cause the CPU to get busy. Maybe a minimal Sleep should be added here.